scripts/odbpy/power_utils.py

```diff
@@ -78,41 +81,39 @@ def find_power_ground_port(port_name, ports):
                 return port
         return None
 
-    reader = OdbReader(input_lef, input_def)
-
     print(f"Top-level design name: {reader.name}")
 
     VDD_PORTS, GND_PORTS = get_power_ground_ports(reader.block.getBTerms())
-    assert (
-        VDD_PORTS and GND_PORTS
-    ), "No power ports found at the top-level. Make sure that they exist and have the USE POWER|GROUND property or they match the arguments specified with --power-port and --ground-port."
+    if len(VDD_PORTS) == 0 or len(GND_PORTS) == 0:
+        print(
+            "No power ports found at the top-level. Make sure that they exist and have the USE POWER|GROUND property or they match the arguments specified with --power-port and --ground-port.",
+            file=sys.stderr,
+        )
+        exit(os.EX_DATAERR)
 
-    vdd_net_idx = None
-    for index, port in enumerate(VDD_PORTS):
+    vdd_net = None
+    gnd_net = None
+    for port in VDD_PORTS + GND_PORTS:
         if port.getNet().getName() == power_port:
-            vdd_net_idx = index
-
-    gnd_net_idx = None
-    for index, port in enumerate(GND_PORTS):
-        if port.getNet().getName() == ground_port:
-            gnd_net_idx = index
-
-    assert (
-        vdd_net_idx is not None
-    ), "Can't find power net at the top-level. Make sure that argument specified with --power-port"
-
-    assert (
-        gnd_net_idx is not None
-    ), "Can't find ground net at the top-level. Make sure that argument specified with --ground-port"
-
-    DEFAULT_VDD = VDD_PORTS[vdd_net_idx].getNet()
-    DEFAULT_GND = GND_PORTS[gnd_net_idx].getNet()
-
-    print("Default power net: ", DEFAULT_VDD.getName())
-    print("Default ground net:", DEFAULT_GND.getName())
-
-    print("Found a total of", len(VDD_PORTS), "power ports.")
-    print("Found a total of", len(GND_PORTS), "ground ports.")
+            vdd_net = port.getNet()
+        elif port.getNet().getName() == ground_port:
+            gnd_net = port.getNet()
+
+    nets_not_found = False
+    if vdd_net is None:
+        print(f"Power port {power_port} not found in design.", file=sys.stderr)
+        nets_not_found = True
+    if gnd_net is None:
+        print(f"Ground port {ground_port} not found in design.", file=sys.stderr)
+        nets_not_found = True
+    if nets_not_found:
+        exit(0)
```

Comment:
that's a zero exit for ,i suppose, an error

Reply:
You are right